src/reducers/sources.js

```diff
@@ -174,8 +188,13 @@ function updateSource(state: Record<SourcesState>, source: Source | Object) {
   if (!source.id) {
     return state;
   }
+  const existingSource = state.getIn(["sources", source.id]);
 
-  return state.mergeIn(["sources", source.id], source);
+  if (existingSource) {
+    const updatedSource = existingSource.merge(source);
+    return state.setIn(["sources", source.id], updatedSource);
+  }
+  return state.mergeIn(["sources", source.id], new SourceRecordClass(source));
```

Comment:
this converts the new Record into a Map, see this [immutable bug](https://github.com/facebook/immutable-js/issues/975). This causes the failures to retrieve attributes without using `get()`.

Reply:
Thanks! is there a good work around?

Reply:
hey @jasonLaster... I changed `state.mergeIn()` in L197 to:
`state.setIn(["sources", source.id], new SourceRecordClass(source);`
and every `source.get("x")` with `source.x` and all tests now pass! I think @DavideDeCenzo may try to finalize it

Reply:
oh thanks. that's promising.  Re-opening